pkg/flow/internal/controller/component.go

```diff
@@ -99,7 +101,7 @@ var (
 
 // NewComponentNode creates a new ComponentNode from an initial ast.BlockStmt.
 // The underlying managed component isn't created until Evaluate is called.
-func NewComponentNode(globals ComponentGlobals, b *ast.BlockStmt) *ComponentNode {
+func NewComponentNode(globals ComponentGlobals, b *ast.BlockStmt, cm *componentMetrics) *ComponentNode {
```

Comment:
nit: don't pass unexported structs to an exported method. I think you can export componentMetrics for now.